### PR TITLE
Adds support for Swift Package Manager

### PR DESCRIPTION
**Pull request by boek**
This was able to build using `swift build` and pull in using Xcode

image

### CODE REVIEW

File: Package.swift [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+ SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import PackageDescription
+
+let package = Package(
+    name: "GCDWebServer",
+    platforms: [.macOS(.v10_10),
+      .iOS(.v9),
+      .tvOS(.v9)],
```
Comment (boek) [inline Package.swift:35]: Not actually sure if this supports tvOS. But it was part of the init template so I left it..

**Comment by boek** [comment]
The more I'm reading about Swift Package Manager it sounds like there may be more steps? I'll report back by the end of the day if this needs anything else

**Comment by SlaunchaMan** [comment]
@boek I attempted this with #475 and it was rejected; I would love to see this move forward, however!

**Comment by boek** [comment]
@SlaunchaMan me too. I almost have a solution that works with very minimal changes, but I have one circular dependency to resolve. I'll update when I have something that runs!

**Comment by nostradani** [comment]
This PR indeed is missing quite some things. This includes public headers and handling of the bundle resources. 
I created another PR instead: https://github.com/swisspol/GCDWebServer/pull/516

**Pull request closed without merging**